### PR TITLE
Added workflow to run pytest automatically

### PR DESCRIPTION
**Pull request by MntiKor**

### CODE REVIEW

**Comment by finetjul** [comment]
Please squash all your commits

**Comment by pchoisel** [comment]
@MntiKor Please squash all your commits

**Comment by MntiKor** [comment]
> @MntiKor Please squash all your commits

Done.

File: .github/workflows/tester.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+name: unit-tests
```
Comment (pchoisel) [inline .github/workflows/tester.yml:1]: Could you add caching on the `env` folder so we don't need to recreated the environment and install pytest each time we run the tests ?

**Review by pchoisel — CHANGES_REQUESTED** [review]

File: .github/workflows/tester.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.11.3'
+      - run: python -m venv env
+      - run: source env/bin/activate
+      - run: env/bin/pip install -e .
+      - run: env/bin/pip install pytest
+      - run: env/bin/pytest
```
[the remark this answers is not in the document]
Comment (MntiKor) [reply .github/workflows/tester.yml:18]: It appears I do need env/bin/, it doesnt work without

**Comment by pchoisel** [comment]
LGTM
Thanks !

**Pull request merged**